components/automate-cli/cmd/chef-automate/verify.go

```diff
@@ -0,0 +1,229 @@
+	}
+
+	if flagsObj.haAWSProvision {
+		err = v.verifyHaAWSProvision(configPath)
+	} else if flagsObj.haAWSManagedProvision {
+		err = v.verifyHaAWSManagedProvision(configPath)
+	} else if flagsObj.haAWSDeploy {
+		err = v.verifyHaAWSDeploy(configPath)
+	} else if flagsObj.haAWSManagedDeploy {
+		err = v.verifyHaAWSManagedDeploy(configPath)
+	} else if flagsObj.standaloneDeploy {
+		err = v.verifyStandaloneDeploy(configPath)
+	} else if flagsObj.haOnpremDeploy {
+		err = v.verifyHaOnpremDeploy(configPath)
+	} else if flagsObj.haOnPremAWSManagedDeploy {
+		err = v.verifyHaOnPremAWSManagedDeploy(configPath)
+	} else if flagsObj.haOnPremCustManagedDeploy {
+		err = v.verifyHaOnPremCustManagedDeploy(configPath)
+	} else if flagsObj.certificates {
+		err = v.Verification.VerifyCertificates("")
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *verifyCmdFlow) verifyHaAWSProvision(configPath string) error {
+	if v.A2HARBFileExist {
+		return status.New(status.InvalidCommandArgsError, "Setup is already Provisioned. Please use --ha-aws-deploy flag.")
+	}
+	err := v.Verification.VerifyHAAWSProvision(configPath)
+	if err != nil {
+		return status.Annotate(err, status.ConfigError)
+	}
+
+	return nil
+}
+
+func (v *verifyCmdFlow) verifyHaAWSManagedProvision(configPath string) error {
+	if v.A2HARBFileExist {
+		return status.New(status.InvalidCommandArgsError, "Setup is already Provisioned. Please use --ha-aws-managed-deploy flag.")
+	}
+	err := v.Verification.VerifyHAAWSManagedProvision(configPath)
+	if err != nil {
+		return status.Annotate(err, status.ConfigError)
+	}
+
+	return nil
+}
+
+func (v *verifyCmdFlow) verifyHaAWSDeploy(configPath string) error {
+	if !v.ManagedServicesOn {
+		if v.A2HARBFileExist {
+			err := v.Verification.VerifyHAAWSDeployment(configPath)
+			if err != nil {
+				return status.Annotate(err, status.ConfigError)
+			}
+			return nil
+		}
+		return status.New(status.InvalidCommandArgsError, errProvisonInfra)
+	}
+	return status.New(status.InvalidCommandArgsError, "This flag will not verify the Managed Services Setup. Please use the --ha-aws-managed-deploy flag.")
+}
+
+func (v *verifyCmdFlow) verifyHaAWSManagedDeploy(configPath string) error {
+	if v.ManagedServicesOn {
+		if v.A2HARBFileExist {
+			err := v.Verification.VerifyHAAWSManagedDeployment(configPath)
+			if err != nil {
+				return status.Annotate(err, status.ConfigError)
+			}
+			return nil
+		}
+		return status.New(status.InvalidCommandArgsError, errProvisonInfra)
+	}
+	return status.New(status.InvalidCommandArgsError, "Managed Services flag is not set. Cannot verify the config.")
+}
+
+func (v *verifyCmdFlow) verifyStandaloneDeploy(configPath string) error {
+	if v.A2HARBFileExist {
+		return status.New(status.InvalidCommandArgsError, "Deployment type does not match with the requested flag.")
+	}
+	err := v.Verification.VerifyStandaloneDeployment(configPath)
+	if err != nil {
+		return status.Annotate(err, status.ConfigError)
+	}
+
```

Comment:
above statement can be written in a single-line statement
`if err := v.Verification.VerifyStandaloneDeployment(configPath); err != nil { return status.Annotate(err, status.ConfigError) } `

Reply:
Done